server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -741,11 +750,13 @@ public XContentBuilder snapshotsToXContent(final XContentBuilder builder, final
                 builder.endObject();
             }
             final IndexVersion version = snapshotDetails.getVersion();
-            if (version != null) {
-                if (version.before(IndexVersion.V_8_10_0)) {
-                    builder.field(VERSION, Version.fromId(version.id()).toString());
+            if (version != null && version.id() != NUMERIC_INDEX_VERSION_MARKER.id()) {
```

Comment:
Given your note on another channel, I think we want this to also output the version if it is the marker version, since otherwise we risk the code [here](https://github.com/elastic/elasticsearch/blob/f77c16b78bab5e9f3ee24974aeba408e254e8c94/server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java#L2141-L2150) interpreting it as 7.5?

Reply:
In practice this should almost always be fine because we re-read these versions before writing the new RepositoryData, but you're right that this is a best-effort process so if it fails we could indeed have problems here. Fixed in fbd2481bf67e30e692eda8c6193ba0eddbce86e0, but I included some suitable logging because I think we want to know about it in case we've missed something in our reasoning.